banjo/gen/llvm/generator.cpp

```diff
@@ -1101,6 +1139,51 @@ Generator::gen(Function_decl const& d)
   fn = nullptr;
 }
 
+// TODO: Code gen coroutines
+void
+Generator::gen(Coroutine_decl const& d)
+{
+  String name = get_name(d);
+  llvm::Type* type = get_type(d.type());
+  lingo_unimplemented("Coroutines");
+}
+
+void
+Generator::gen(Class_decl const& d)
+{
+  // Check if class has already been declared
+  if (types.lookup(&d))
+    return;
+
+  // Holds the types and also the super classes.
+  // Super classes should be put at the head of the vector
+  std::vector<llvm::Type*> ts;
+
+  // Check for parent classes and add them to ts
+
+  // Add members to ts
+
```

Comment:
Since the Class_def only hods a list of stamens which consists of both member variables and method declarations it is not easy to separate them like in beaker. Not sure if we need to change how the definition is held or if there is a way to do it here.